Userland/Games/Spider/Game.cpp

```diff
@@ -0,0 +1,320 @@
+            m_new_game_animation_delay = 0;
+            auto& current_pile = stack(piles.at(m_new_game_animation_pile));
+
+            // for first 4 piles, draw 6 cards
+            // for last 6 piles, draw 5 cards
+            size_t cards_to_draw = m_new_game_animation_pile < 4 ? 6 : 5;
+
+            if (current_pile.count() < (cards_to_draw - 1)) {
+                auto card = m_new_deck.take_last();
+                card->set_upside_down(true);
+                current_pile.push(card);
+            } else {
+                current_pile.push(m_new_deck.take_last());
+                ++m_new_game_animation_pile;
+            }
+
+            if (m_new_game_animation_pile == piles.size()) {
+                VERIFY(m_new_deck.size() == 50);
+                while (!m_new_deck.is_empty())
+                    stack(Stock).push(m_new_deck.take_last());
+                m_new_game_animation = false;
+                m_waiting_for_new_game = true;
+                stop_timer();
+            }
+        }
+    }
+
+    if (!m_focused_cards.is_empty()) {
+        for (auto& focused_card : m_focused_cards)
+            focused_card.clear(painter, s_background_color);
+    }
+
+    for (auto& stack : m_stacks) {
+        stack.draw(painter, s_background_color);
+    }
+
+    if (!m_focused_cards.is_empty()) {
+        for (auto& focused_card : m_focused_cards) {
+            focused_card.draw(painter);
+            focused_card.save_old_position();
+        }
+    }
+
+    if (!m_mouse_down) {
+        if (!m_focused_cards.is_empty()) {
+            for (auto& card : m_focused_cards)
+                card.set_moving(false);
+            m_focused_cards.clear();
+        }
+
+        if (m_focused_stack) {
+            m_focused_stack->set_focused(false);
+            m_focused_stack = nullptr;
+        }
+    }
+}
+
+void Game::mousedown_event(GUI::MouseEvent& event)
+{
+    GUI::Frame::mousedown_event(event);
+
+    if (m_new_game_animation)
+        return;
+
+    auto click_location = event.position();
+    for (auto& to_check : m_stacks) {
+        if (to_check.type() == CardStack::Type::Waste)
+            continue;
+
+        if (to_check.bounding_box().contains(click_location)) {
+            if (to_check.type() == CardStack::Type::Stock) {
+                draw_cards();
+            } else if (!to_check.is_empty()) {
+                auto& top_card = to_check.peek();
+
+                if (top_card.is_upside_down()) {
+                    if (top_card.rect().contains(click_location)) {
+                        top_card.set_upside_down(false);
+                        start_timer_if_necessary();
+                        update(top_card.rect());
+                    }
+                } else if (m_focused_cards.is_empty()) {
+                    to_check.add_all_grabbed_cards(click_location, m_focused_cards, Cards::CardStack::Same);
+                    m_mouse_down_location = click_location;
+                    to_check.set_focused(true);
```

Comment:
```suggestion
                    if (m_focused_stack)
                        m_focused_stack->set_focused(false);
                    to_check.set_focused(true);
```

I don't know how exactly but I somehow end up here with `m_focused_stack` being non-null. At least for me this fixes the problem where I can't drag&drop cards.

Reply:
FWIW I've been able to reproduce that problem pretty much every game. I just can't tell you what it is I'm doing that triggers it. :)